### PR TITLE
fix(iroh-net): Fix some flaky magicsock tests

### PR DESCRIPTION
**Pull request by flub**
## Description

Fix flaky tests in iroh-net (hopefully).  The main issue was that the
testing code meshing together the sockets had a race-condition and
would sometimes fail.

Other improvements:

- Logging improvements of various parts.  These tests should now
  provide better info on failure.

- Add alternative formatting of NodeId to use the short format.

- Use NodeId in the public API, rather than PublicKey.

- Move multi-threaded logging setup to the testing utils crate now
  that it can work properly.  More code will have to switch.

## Notes & open questions

The local_endpoints_change API is really difficult to use correctly.
Even the way this testing code uses it now is wrong.  Instead of the
two current APIs there should be a single API which returns a stream.
I'll do this as a separate followup PR though.

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [x] Tests if relevant.

### CODE REVIEW

File: iroh-net/src/key.rs [hunk 1]
```diff
@@ -227,7 +227,11 @@ impl Debug for PublicKey {
 
 impl Display for PublicKey {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", base32::fmt(self.as_bytes()))
+        if f.alternate() {
```
Comment (dignifiedquire) [inline iroh-net/src/key.rs:230]: when/how is this triggered?

Comment (dignifiedquire) [reply iroh-net/src/key.rs:230]: shouldn’t it be the other way around then,? # prints longer more detailed messages

Comment (flub) [reply iroh-net/src/key.rs:230]: I could be rather tempted to switch this around and do the short format by default and the long format for alternate.  That would reduce rather a lot of verbosity in our code.

Comment (dignifiedquire) [reply iroh-net/src/key.rs:230]: the issue is that, even externally, we rely on `to_string()` roundtriping

Comment (flub) [reply iroh-net/src/key.rs:230]: yeah, it'd probably break things if this can't be round-tripped.  so maybe we'll have to settle for this.

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**